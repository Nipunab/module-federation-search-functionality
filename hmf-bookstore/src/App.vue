<template>

  <div class="nb-container-fluid">
    <transition :name="transitionName">
      <router-view />
    </transition>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component({
  name: 'App',
  components: {},
})
export default class AppComponent extends Vue {
  public transitionName = 'slide-fade';
}
</script>

<style lang="scss">
@import 'assets/styles/index.scss';
</style>
