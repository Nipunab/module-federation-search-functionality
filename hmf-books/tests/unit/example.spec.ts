import { shallowMount, createLocalVue, Wrapper } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import { attest, toHaveZeroViolations } from 'attest-jest';

import App from '../../src/App.vue';

expect.extend(toHaveZeroViolations);

const slotComponent = {
  name: 'slot-component',
  props: ['text'],
  template: '<span>{{ text }}</span>',
};
const stubs = {
  'router-view': true,
  Module: slotComponent,
};

describe('App.vue', () => {
  let wrapper: Wrapper<App>;

  beforeEach(() => {
    const localVue = createLocalVue();
    wrapper = shallowMount(App, {
      stubs,
      mocks: {
        $route: '/',
      },
      localVue,
    });
  });

  afterEach(() => {
    if (wrapper) {
      wrapper.vm.$destroy();
    }
  });

  it('Matches Snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Has no accessibility violations', (done) => {
    renderToString(App)
      .then((str) => attest(str, { rules: { region: { enabled: false } } }))
      .then((results) => {
        expect(results).toHaveZeroViolations();
        done();
      });
  });
});
