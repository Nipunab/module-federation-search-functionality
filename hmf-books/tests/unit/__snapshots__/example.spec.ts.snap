// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Matches Snapshot 1`] = `
<div
  class="nb-container-fluid"
>
  <transition-stub
    name="slide-fade"
  >
    <router-view-stub />
  </transition-stub>
</div>
`;
